src/main/scala/chisel3/util/experimental/CIRCTSRAMInterface.scala

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util.experimental
+
+import chisel3._
+import chisel3.experimental.BaseModule
+import chisel3.experimental.hierarchy.Instance
+import chisel3.util.log2Ceil
+
+import scala.collection.immutable.SeqMap
+
+object CIRCTSRAMParameter {
+  implicit val rw: upickle.default.ReadWriter[CIRCTSRAMParameter] = upickle.default.macroRW
+}
+
+case class CIRCTSRAMParameter(
+  moduleName:      String,
+  read:            Int,
+  write:           Int,
+  readwrite:       Int,
+  depth:           Int,
+  width:           Int,
+  maskGranularity: Int) {
+  def masked: Boolean = maskGranularity != 0
+}
+
+class CIRCTSRAMInterface(memoryParameter: CIRCTSRAMParameter) extends Record {
+
+  class R extends Record {
+    lazy val clock = Input(Clock())
+    lazy val address = Input(UInt(log2Ceil(memoryParameter.depth).W))
+    lazy val data = Output(UInt(memoryParameter.width.W))
+    lazy val enable = Input(Bool())
+
+    val elements: SeqMap[String, Data] = SeqMap(
+      "clk" -> clock,
+      "addr" -> address,
+      "data" -> data,
+      "en" -> enable
+    )
+  }
```

Comment:
Do we actually want these to be inner classes? What if a user wants to represent one of these ports as just a `Wire`?